src/main/java/catan/services/util/play/MainStageUtil.java

```diff
@@ -30,6 +35,17 @@ public void updateNextMove(GameBean game) {
         game.setCurrentMove(nextMoveNumber);
     }
 
+    public void produceResources(GameBean game) {
+        for (HexBean hex : game.getHexesWithCurrentDiceValue()) {
+            for (NodeBean node : hex.getNodesWithBuildings()) {
+                Resources userResources = node.getBuilding().getBuildingOwner().getResources();
```

Comment:
intorduce one more variable for 'node.getBuilding()' that is called more than once, as you use it in very long constructions